server/controllers/request.js

```diff
@@ -0,0 +1,108 @@
+const Request = require("../models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /request/
+// @desc create a new request by owner
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  try {
+    const user = await User.findById(req.user.id);
+    if (!user) {
+      return res
+        .status(401)
+        .json({ message: 'Not authorized' });
+    }
+
+    const { sitterId, startDate, endDate } = req.body;
+    const requestModel = {
+        ownerId: user._id, 
+        sitterId,
+        startDate,
+        endDate,
+    };
+
+    const requestExists = await Request.findOne(requestModel);
+    if (requestExists) {
+      return res
+          .status(400) 
+          .json({ message: 'You have already made a request' });
+    }
+
+    const sitterRequest = await Request.create(requestModel); 
+    res.status(201).json({ sitterRequest });
+  } catch (error) {
+    next(error);
+  } 
+  
+});
+
+// @route GET /request
+// @desc list of requests made for logged in user
+// @access Private
+exports.userRequests = asyncHandler(async (req, res, next) => {
+  try {
+    const user = await User.findById(req.user.id);
+    if (!user) {
+      return res
+        .status(401)
+        .json({ message: 'Not authorized' });
+    }
+    
+    const requestsByUser = await Request.find({ 
+      $or: [
+        { ownerId: user._id },
+        { sitterId: user._id }
+      ] 
+    });
+    if (!requestsByUser) {
+      return res
+        .status(200)
+        .json({ message: "There are no requests"});
+    }
+
+      res.status(200).json({ requestsByUser });
```

Comment:
As an alternative approach, you could remove this if statement and send the empty array as usual since an empty array would imply that "there are no requests". Feel free to go with the approach you prefer.

Reply:
done